### PR TITLE
Adding in CPU time to wherever time is defined as variable "cpu_time"

### PR DESCRIPTION
**Pull request by thijssnelleman**
Changed the TrialValue to hold CPU time and have SMBO use this value to calculate its budgets

- Same behaviour as the original 'time' variable, except in its calculation tries to estimate CPU time with process_time instead.
- Added in to TrialValue object, RunHistory and SMBO so it can be recorded and correctly calculate the remaining CPU budget

Pytest are passing (on my machine)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
 ## Improvements
 - Add logger information on handling of stopIteration error (#960)
 - Replace deprecated ConfigSpace methods (#1139)
+- Separated Wallclock time measurements from CPU time measurements and storing them under new 'cpu_time' variable (#????)
```
Comment (benjamc) [inline CHANGELOG.md:13]: update PR number

File: smac/main/smbo.py [hunk 2]
```diff
@@ -137,6 +138,11 @@ def used_target_function_walltime(self) -> float:
         """Returns how much walltime the target function spend so far."""
         return self._used_target_function_walltime
 
+    @property
+    def used_target_function_cputime(self) -> float:
+        """Returns how much cputime the target function spend so far."""
```
Comment (benjamc) [inline smac/main/smbo.py:143]: describe that CPU time generally means time spent on the hardware and not just CPU (whether its CPU, TPU, GPU or whatever)

File: smac/runhistory/dataclasses.py [hunk 3]
```diff
@@ -98,6 +98,7 @@ class TrialValue:
     ----------
     cost : float | list[float]
     time : float, defaults to 0.0
+    cpu_time : float, defaults to 0.0
```
Comment (benjamc) [inline smac/runhistory/dataclasses.py:101]: describe CPU time (what it means and why/ in what way it differs from time)

File: smac/runhistory/runhistory.py [hunk 4]
```diff
@@ -191,6 +192,8 @@ def add(
             Cost of the evaluated trial. Might be a list in case of multi-objective.
         time : float
             How much time was needed to evaluate the trial.
+        cpu_time : float
```
Comment (benjamc) [inline smac/runhistory/runhistory.py:195]: Same here, add a little more info

File: smac/runhistory/runhistory.py [hunk 5]
```diff
@@ -848,6 +856,7 @@ def load(self, filename: str | Path, configspace: ConfigurationSpace) -> None:
         self._n_id = len(self._config_ids)
 
         # Important to use add method to use all data structure correctly
+        # NOTE: These hardcoded indices can easily lead to trouble
```
Comment (benjamc) [inline smac/runhistory/runhistory.py:859]: agreed

**Review by benjamc — CHANGES_REQUESTED** [review]
is there any way to test the new cpu return? if yes, please add a little test

otherwise, codewise the PR LGTM, it only needs a bit more documentation.
You can basically copy/paste the description for cpu_time everywhere.

File: smac/runhistory/runhistory.py (hunk 5 above)
Comment (thijssnelleman) [reply smac/runhistory/runhistory.py:859]: https://github.com/automl/SMAC3/issues/1174

**Comment by thijssnelleman** [comment]
Added in CPU time tests in target function runner tests

**Review by benjamc — APPROVED** [review]

**Pull request merged**